engine/src/main/scala/cromwell/engine/backend/CromwellBackend.scala

```diff
@@ -15,24 +15,43 @@ import org.slf4j.LoggerFactory
   */
 object CromwellBackend {
   private val log = LoggerFactory.getLogger(getClass.getName)
-  private var _backend: Option[Backend] = None
+  private var _backend: Option[Map[String, Backend]] = None
+  private var _defaultBackend: Option[Backend] = None
 
-  def initBackend(backendType: String, actorSystem: ActorSystem): Backend = {
+  def initBackends(backendTypes: List[String], defaultBackend: String, actorSystem: ActorSystem) = {
+
+    _backend = Option((backendTypes map { backendType => (backendType, Backend.from(backendType, actorSystem)) }).toMap)
+    _defaultBackend = Option(Backend.from(defaultBackend, actorSystem))
+
+//    _backend match {
+//      case None =>
+//        _backend = Option((backendTypes map { backendType => (backendType, Backend.from(backendType, actorSystem)) }).toMap)
+//        _defaultBackend = Option(Backend.from(defaultBackend, actorSystem))
+//      case Some(x) if backendTypes.equals(x.keys.toList) => // TODO: This is only here because Main creates two instances of WorkflowManagerSystem. We should probably fix that.
+//      case Some(x) => throw new IllegalStateException(s"The set of backends has already been initialised to ${x.mkString}. Don't do that again (with ${backendTypes.mkString})!")
+//    }
```

Comment:
Yeah, so in theory this check would be useful. In reality, initBackends is called so many times that this test causes problems. Hopefully in the future, we won't have to init backends at all :grinning: